### PR TITLE
ARXIVNG-1440 tweaks toward functionality to support submission UI

### PR DESCRIPTION
**Pull request by erickpeirson**
Most of these are minor tweaks/changes, and documentation. The most significant change is adding some additional semantics for failures, i.e. adding ``Reason`` to the domain.

This also adds a bunch of tests. Not quite up to snuff for ARXIVNG-1334, but getting there.

### CODE REVIEW

**Comment by erickpeirson** [comment]
This is getting pretty stable for 0.1. Probably a few more minor changes. Still really just targets v0.1 support (i.e. not a lot of attention to different formats).

File: compiler/compiler.py [hunk 1]
```diff
@@ -317,8 +348,11 @@ def compile_source(source: SourcePackage,
         args.append(f'-U {tex_tree_timestamp}')
 
     logger.debug('run image %s with args %s', image, args)
-    run_docker(image, args=args,
-               volumes=[(host_source_dir, '/autotex')])
+    code, stdout, stderr = run_docker(image, args=args,
+                                      volumes=[(host_source_dir, '/autotex')])
+    # TODO: not sure about this.
```
Comment (JaimieMurdock) [inline compiler/compiler.py]: Not sure about this in a "exclude commented lines" way or in a "I want to include these lines but haven't tested" way?

I think that looks reasonable.

File: compiler/compiler.py [hunk 2]
```diff
@@ -262,41 +300,34 @@ def _store_compilation_result(status: CompilationStatus,
...
-    logger.debug('container_source_root: %s', container_source_root)
-    logger.debug('leaf_path: %s', leaf_path)
-    logger.debug('host_source_dir: %s', host_source_dir)
-    logger.debug('got image %s', image)
 
     args = [
         '-S /autotex',
-        f'-p {source.source_id}',
-        f'-f {output_format}',  # This doesn't do what we think it does.
+        f'-p 1901.00123',   # {source.source_id}
```
Comment (JaimieMurdock) [inline compiler/compiler.py]: Note that this needs to be changed later - I'd flag with a `# TODO:` so we don't miss it!

File: compiler/compiler.py [hunk 3]
```diff
@@ -329,22 +363,23 @@ def compile_source(source: SourcePackage,
         # version affix). But at the end of the day there should be only one
         # file in the format that we requested, so that's as specific as we
         # should need to be.
-        cache_results = [fp for fp in os.listdir(cache) if fp.endswith(f'.{ext}')]
+        cache_results = [f for f in os.listdir(cache) if f.endswith(f'.{ext}')]
```
Comment (JaimieMurdock) [inline compiler/compiler.py]: good semantics

File: compiler/compiler.py [hunk 4]
```diff
@@ -329,22 +363,23 @@ def compile_source(source: SourcePackage,
...
-    return output_path, tex_log_path if os.path.exists(tex_log_path) else None
+    tex_log_path = os.path.join(source_dir, 'tex_logs', 'autotex.log')
+    return out_path, tex_log_path if os.path.exists(tex_log_path) else None
 
 
-def run_docker(image: str, volumes: list = [], ports: list = [],
-               args: List[str] = [], daemon: bool = False) -> None:
+def run_docker(image: str, volumes: list = [],
+               ports: list = [], args: List[str] = [],
+               daemon: bool = False) -> Tuple[int, str, str]:
```
Comment (JaimieMurdock) [inline compiler/compiler.py]: perfect on returning actual info about process.

File: compiler/compiler.py (hunk 1 above)
Comment (erickpeirson) [reply compiler/compiler.py]: I originally thought this was indicative of a problem with the tarball. Now I'm not so sure. Seems to be fine, but will continue to investigate.

**Pull request merged**